### ISSUE TITLE
Problem with PROXY download 5.7.0 (Does not wait for the request to end??)

### ISSUE THREAD

**Issue by MaciejGrymuza** [selenide/selenide#1065]
I have noticed 2 random behaviours when downloading ZIP files with PROXY. It looks like when Selenide downloads bigger ZIP it does not wait for the download request to finish. So when Selenide downloads some ZIP, and later we try to unzip that file (only with java code), the operation fails because "ZIP is empty". 
1. In location set by user, Selenide creates folder where downloaded file should be, but:
- there is some file in that folder, but definitely not the correct one
- ZIP file is present in the location that was set  by user (Configuration.downloadsFolder) or the default folder. See attached files "DownloadedFiles1" and "ContentInsideNewFolder"

![DownloadedFiles1](https://user-images.githubusercontent.com/50072501/74641000-d7630800-5170-11ea-838c-bc082dd2734e.png)
![ContentInsideNewFolder](https://user-images.githubusercontent.com/50072501/74641004-d8943500-5170-11ea-91ed-eb7a6e5562b6.png)

2. Sometimes Selenide only creates a new folder in location set by user and there is no ZIP file at all. Check screens "NoZIP" 
Maybe this log will tell something: 
11:39:20.324 [LittleProxy-4-ProxyToServerWorker-4] INFO com.codeborne.selenide.proxy.FileDownloadFilter - Cannot extract file name from http headers. Found headers: [list of headers]

![NoZIP](https://user-images.githubusercontent.com/50072501/74641277-4f313280-5171-11ea-8c65-de314872cafe.png)

3. Sometimes in the folder created by Selenide there is some HTML element downloaded. Like .svg file in attached "StrangeElement" screen. 

![StrangeElement](https://user-images.githubusercontent.com/50072501/74641827-3e34f100-5172-11ea-986b-85c15f89423c.png)

ENVIRONMENT:
Chrome 79.0
Win 10
Selenide 5.7.0

Configuration.proxyEnabled = true;
Configuration.fileDownload = PROXY;

**Comment by asolntsev** [comment]
Hi @MaciejGrymuza 
1. Yes, Selenide intercepts ALL responses that browser receives after click. There can be many of them, and some of them might not be related to your downloading. For example, sometimes Chrome loads some images on background or checks for updates.
2. Selenide tries to detect which on is the right download. Sometimes it detects incorrectly. 
3. I am working on feature to fix this issue. I am going to add an overloaded method `$.download(FileFilter filter)`. Using it, you will be able to explicitly set file name, extension or some other criteria. Like 

```java
  File cv = $("input").download(withFilename("blah.pdf"));
  File cv = $("input").download(withFilenameMatching(".*\\.xls"));
```

4. I never saw that Selenide downloaded an empty file, or created an empty folder. How can I reproduce it?

**Comment by asolntsev** [comment]
Solved in Selenide 5.9.0 by PR #1080

**Comment by MaciejGrymuza** [comment]
@asolntsev this is how it goes. First step is download operation. We are downloading ZIP file, that contains 7 PDF documents inside:
![1](https://user-images.githubusercontent.com/50072501/76001342-3d87b300-5f05-11ea-82c6-cfbeca336178.png)

And this is the result in the downloads folder that I chose for Selenide: 
![image](https://user-images.githubusercontent.com/50072501/76001449-63ad5300-5f05-11ea-899f-b3ad09f220e9.png)
Selenide download some folder. Now let's open that folder:
![image](https://user-images.githubusercontent.com/50072501/76001500-79227d00-5f05-11ea-9e20-7df419874eaa.png)
There is some strange file inside. 
Also, if you cannot reproduce it, maybe it would be good idea to:
1. Download some ZIP file that contains many elements.
2. Perform some action on the downloaded file (like unzipping, reading etc) immidiately after downloading it. So in the code it would be sth like (no "waits" after downloading file):
  File downloaded = $(By.className("action-download")).download();
  downloaded.unzip();
